test/models.jl

```diff
@@ -0,0 +1,164 @@
+module TestModels
+
+  using SearchLight
+
+  ######## Model from Genie-Searchligth-example-app extracted ############
+  export Book, BookWithInterns
+  using SearchLight, Dates
+
+  ######## Model from Genie-Searchligth-example-app extracted ############
+  export Callback
+  export seed, fields_to_store
+
+  mutable struct Book <: AbstractModel
+
+    ### FIELDS
+    id::DbId
+    title::String
+    author::String
+    cover::String
+
+    ### VALIDATION
+    # validator::ModelValidator
+
+    ### CALLBACKS
+    # before_save::Function
+    # after_save::Function
+    # on_save::Function
+    # on_find::Function
+    # after_find::Function
+
+    ### SCOPES
+    # scopes::Dict{Symbol,Vector{SearchLight.SQLWhereEntity}}
+
+    ### constructor
+    Book(;
+      ### FIELDS
+      id = DbId(),
+      title = "",
+      author = "",
+      cover = "",
+
+      ### VALIDATION
+      # validator = ModelValidator([
+      #   ValidationRule(:title, BooksValidator.not_empty)
+      # ]),
+
+      ### CALLBACKS
+      # before_save = (m::Todo) -> begin
+      #   @info "Before save"
+      # end,
+      # after_save = (m::Todo) -> begin
+      #   @info "After save"
+      # end,
+      # on_save = (m::Todo, field::Symbol, value::Any) -> begin
+      #   @info "On save"
+      # end,
+      # on_find = (m::Todo, field::Symbol, value::Any) -> begin
+      #   @info "On find"
+      # end,
+      # after_find = (m::Todo) -> begin
+      #   @info "After find"
+      # end,
+
+      ### SCOPES
+      # scopes = Dict{Symbol,Vector{SearchLight.SQLWhereEntity}}()
+
+    ) = new(id, title, author, cover                                                             ### FIELDS
+            # validator,                                                                  ### VALIDATION
+            # before_save, after_save, on_save, on_find, after_find                       ### CALLBACKS
+            # scopes                                                                      ### SCOPES
+            )
+  end
+
+  mutable struct BookWithInterns <: AbstractModel
+    ### INTERNALS
```

Comment:
Internal fields should no longer be needed with the latest API - can you please remove?

Reply:
changed

Reply:
The internal (the ones starting with `_`) are still here though...